plugins/org.yakindu.sct.generator.cpp/src/org/yakindu/sct/generator/cpp/templates/ClassDeclaration.xtend

```diff
@@ -0,0 +1,127 @@
+package org.yakindu.sct.generator.cpp.templates
+
+import java.util.List
+
+class ClassDeclaration {
+	protected CharSequence name
+	protected CharSequence comment
+	
+	protected List<CharSequence> superTypes = newArrayList
+	
+	protected List<CharSequence> publicMembers = newArrayList
+	protected List<CharSequence> protectedMembers = newArrayList
+	protected List<CharSequence> privateMembers = newArrayList
+	
+	def ClassDeclaration name(CharSequence name) {
```

Comment:
setter should be named 'setName'. This is cleaner from a Java/Xtend perspective and better supported by Xtend. So initialized of ClassDeclaration instances can be written more nicely.